pkg/webhook/server.go

```diff
@@ -82,6 +86,74 @@ func (s *Server) setDefaults() {
 	}
 }
 
+// For registers defaulting and validation webhooks for the provided types that implement either
+// webhooktuil.validator or webhookutil.defaulter.
+func (s *Server) EnableDefaultsFor(scheme *runtime.Scheme, objects ...runtime.Object) error {
+	// Create a webhook for each type
+	for _, object := range objects {
+		wh := newMutatingWebhookFor(object)
+		if wh == nil {
+			continue
+		}
+
+		gvks, _, err := scheme.ObjectKinds(object)
+		if err != nil {
+			return err
+		}
+		if len(gvks) != 1 {
+			return fmt.Errorf("expected only GVK returned by scheme.ObjectKinds")
```

Comment:
we should some webhook related context to the error message and also include all the GVKs to help in debugging.